docs/pages/principles.rst

```diff
@@ -0,0 +1,89 @@
+Principles
+==========
+
+This section describes a few elements which are at the core of ``Collection``, 
+and provides guidance on how to get the most out of the package.
+
+Immutability
+------------
+
+Most :ref:`operations <Methods (operations)>` return a new collection object
+rather than modifying the original one. This makes your code easier to reason
+about and more testable.
+
+A few notable exceptions to this are methods which return scalar values like ``count``;
+you can read more about them in the :ref:`operations background <Background>` section. 
+
+Laziness
+--------
+
```

Comment:
`Collection` is "lazy" by default, it's one of it's main principle.
Lazy evaluation, or call-by-need, is an evaluation strategy which delays the evaluation of an expression until its value is really needed.
In `Collection`, the operations are executed on the input stream only when iterating over it or when using very specific operation like: `all` or `squash`.

Reply:
is this a comment meant to modify what's in the **Behaviour** section, or is this something to include in addition to that?

Reply:
This was initially for the "Laziness" paragraph, but I noticed that it was kinda the same in the next paragraph. Should we merge them together?

Reply:
yep I'll see what I can do to merge them